daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala

```diff
@@ -455,6 +497,674 @@ class EvaluationOrderTest extends AnyFreeSpec with Matchers with Inside {
+             """,
+            Array(SParty(alice), SParty(alice)),
+            Set(alice),
+          )
+          inside(res) {
+            case Success(Left(SErrorDamlException(IE.WronglyTypedContract(_, T, Dummy)))) =>
+              msgs shouldBe Seq("starts test")
+          }
+        }
+
+        // TEST_EVIDENCE: Semantics: Evaluation order of exercise of a cached global contract with failure authorization
+        "authorization failures" in {
+          val (res, msgs) = evalUpdateApp(
+            pkgs,
+            e"""\(sig: Party) (obs : Party) (exercisingParty : Party) ->
+                  ubind cId: ContractId M:T <- create @M:T M:T { signatory = sig, observer = obs, precondition = True, key = M:toKey sig, nested = M:buildNested 0 }
+                  in 
+                    Test:exercise_by_id exercisingParty cId (M:Either:Left @Int64 @Int64 0)
+                  """,
+            Array(SParty(alice), SParty(bob), SParty(charlie)),
+            Set(alice, charlie),
+            getContract = getContract,
+          )
+
+          inside(res) { case Success(Left(SErrorDamlException(IE.FailedAuthorization(_, _)))) =>
+            msgs shouldBe Seq(
+              "starts test",
+              "choice controllers",
+              "choice observers",
+            )
+          }
+        }
+      }
+
+      // TEST_EVIDENCE: Semantics: Evaluation order of exercise of an unknown contract
+      "unknown contract" in {
+        val (res, msgs) = evalUpdateApp(
+          pkgs,
+          e"""\(exercisingParty : Party) (cId: ContractId M:T) -> Test:exercise_by_id exercisingParty cId (M:Either:Left @Int64 @Int64 0)""",
+          Array(SParty(alice), SContractId(cId)),
+          Set(alice),
+          getContract = PartialFunction.empty,
+        )
+        inside(res) { case Failure(SpeedyTestLib.UnknownContract(`cId`)) =>
+          msgs shouldBe Seq("starts test", "queries contract")
+        }
+      }
+
+      // TEST_EVIDENCE: Semantics: Evaluation order of exercise with argument exceeding max nesting
+      "argument exceeds max nesting" in {
+        val (res, msgs) = evalUpdateApp(
+          pkgs,
+          e"""\(exercisingParty : Party) (cId: ContractId M:T) -> Test:exercise_by_id exercisingParty cId (M:Either:Left @Int64 @Int64 100)""",
+          Array(SParty(alice), SContractId(cId)),
+          Set(alice),
+          getContract = getContract,
+        )
+        inside(res) { case Success(Left(SErrorDamlException(IE.Limit(IE.Limit.ValueNesting(_))))) =>
+          msgs shouldBe Seq(
+            "starts test",
+            "queries contract",
+            "contract signatories",
+            "contract observers",
+            "key",
+            "maintainers",
+            "choice controllers",
+            "choice observers",
+          )
+        }
+      }
+
+      // TEST_EVIDENCE: Semantics: Evaluation order of exercise with output exceeding max nesting
+      "output exceeds max nesting" in {
```

Comment:
```suggestion
      "result exceeds max nesting" in {
```
I think usually we call it choice result not output.